src/data/matrix/kronecker_product_bialgebra.lean

```diff
@@ -0,0 +1,138 @@
+  (B : matrix n p S) [is_scalar_tower α R β] [is_scalar_tower α S β] :
+  kronecker_biprod α R S β (reindex_linear_equiv α R eₗ eₘ A) B =
+  reindex_linear_equiv α _ (eₗ.prod_congr (equiv.refl _)) (eₘ.prod_congr (equiv.refl _))
+  (kronecker_biprod α R S β A B) := by { ext ⟨i, i'⟩ ⟨j, j'⟩, refl }
+
+lemma kronecker_biprod_reindex_right (eₙ : n ≃ n') (eₚ : p ≃ p') (A : matrix l m R)
+  (B : matrix n p S) [is_scalar_tower α R β] [is_scalar_tower α S β] :
+  kronecker_biprod α R S β A (reindex_linear_equiv α S eₙ eₚ B) =
+  reindex_linear_equiv α _ ((equiv.refl _).prod_congr eₙ) ((equiv.refl _).prod_congr eₚ)
+  (kronecker_biprod α R S β A B) := by { ext ⟨i, i'⟩ ⟨j, j'⟩, refl }
+
+lemma kronecker_biprod_one_one [decidable_eq m] [decidable_eq n] [is_scalar_tower α R β]
+  [is_scalar_tower α S β] : kronecker_biprod α R S β (1 : matrix m m R)
+    (1 : matrix n n S) = (1 : matrix (m × n) (m × n) β) :=
+    by { ext ⟨i, i'⟩ ⟨j, j'⟩, simp [kronecker_biprod, one_apply, algebra_map_eq_smul_one, ite_smul,
+      ite_and] }
+
+theorem kronecker_biprod_mul (A : matrix l m R) (B : matrix m n R) (A' : matrix l' m' S)
+  (B' : matrix m' n' S) [is_scalar_tower α R β] [is_scalar_tower α S β] :
+  kronecker_biprod α R S β (A ⬝ B) (A' ⬝ B') =
+    (kronecker_biprod α R S β A A') ⬝ (kronecker_biprod α R S β B B') :=
+begin
+  ext ⟨i, i'⟩ ⟨j, j'⟩,
+  simp only [mul_apply, kronecker_biprod, algebra_map_eq_smul_one, mul_one, algebra.mul_smul_comm,
+    linear_map.coe_mk, algebra.smul_mul_assoc, ← finset.univ_product_univ, finset.sum_product],
+  simp_rw [finset.sum_smul, finset.smul_sum, ← smul_eq_mul],
+  repeat {apply finset.sum_congr, refl, intros _ _,},
+  rw [is_scalar_tower.smul_assoc, id.smul_eq_mul (A' i' x_1) (B' x_1 j')],
+  simp only [id.smul_eq_mul, ← algebra_map_eq_smul_one, smul_def, ring_hom.map_mul],
+  ring_nf,
+end
+
+theorem kronecker_biprod_assoc {T : Type*} [comm_semiring T] [algebra α T] [algebra T β]
+ [is_scalar_tower α R β] [is_scalar_tower α S β] [is_scalar_tower α T β] [is_scalar_tower α β β]
+ (A : matrix m m' R) (B : matrix n n' S) (C : matrix p p' T) :
+  @matrix.linear_equiv_index_assoc m n p _ _ _ m' n' p' _ _ _ β α _ _ _
+  (kronecker_biprod α β T β (kronecker_biprod α R S β A B) C) =
+    kronecker_biprod α R β β A (kronecker_biprod α S T β B C)
+    :=
+begin
+  simp only [matrix.linear_equiv_index_assoc, kronecker_biprod, linear_map.coe_mk, id.map_eq_self,
+    reindex_apply, linear_equiv.coe_mk],
+  ext i j,
+  simp only [equiv.prod_assoc_symm_apply, reindex_linear_equiv_apply, minor_apply, reindex_apply,
+    mul_assoc],
+end
+
+end matrix_bialgebra
+
+namespace kronecker_product
+
+open algebra matrix matrix_bialgebra
+open_locale matrix
+
+variables (R : Type*) [comm_semiring R]
+variables {l m n p l' m' n' p' : Type*}
+variables [fintype l] [fintype m] [fintype n] [fintype p]
+variables [fintype l'] [fintype m'] [fintype n'] [fintype p']
+
+def kronecker_prod [is_scalar_tower R R R] (A : matrix l m R) (B : matrix n p R) :
```

Comment:
`[is_scalar_tower R R R]` is redundant here, lean will find it for you.

Reply:
Thanks, corrected!